content/techniques/file-upload.md

```diff
@@ -120,7 +120,7 @@ export abstract class FileValidator<TValidationOptions = Record<string, any>> {
 - `MaxFileSizeValidator` - Checks if a given file's size is less than the provided value (measured in `bytes`)
 - `FileTypeValidator` - Checks if a given file's mime-type matches the given value. 
 
-> warning **Warning** The [FileTypeValidator](https://github.com/nestjs/nest/blob/master/packages/common/pipes/file/file-type.validator.ts) class uses a naive implementation to verify the incoming file's type. Consider using a custom implementation (like checking the file's [magic number](https://www.ibm.com/support/pages/what-magic-number)) if your app requires a safer solution.
+> warning **Warning** To verify file type, [FileTypeValidator](https://github.com/nestjs/nest/blob/master/packages/common/pipes/file/file-type.validator.ts) class uses the type as detected by multer. By default, multer derives file type from file extension on user's device. However, it does not check actual file contents. As files can be renamed to arbitraty extensions, consider using a custom implementation (like checking the file's [magic number](https://www.ibm.com/support/pages/what-magic-number)) if your app requires a safer solution.
```

Comment:
I wonder if we can link to `file-type` package here as an example library that can be used exactly to check magic numbers in Buffers, Streams, etc? Or we generally don't want to recommend specific packages?

Reply:
We can leave the research part to the end-user. We try to not recommend specific packages, unless it's really useful/important